### ISSUE TITLE
[BUG][Objectbricks] Too few arguments to function Pimcore\Model\DataObject\AbstractObject::get()

### ISSUE THREAD

**Issue by RuudvdvNB** [pimcore/pimcore#6666]
# Bug Report

### Expected behavior
In Pimcore 6.6.4, when creating a new variant (with inheritance) using the PHP API you can also set values to a new object brick. When omitting the mandatory check, i expect the object to be saved even if i don't fill all the mandatory fields.

### Actual behavior
As soon as you omit the mandatory check inside the object brick there is a inheritance check. If the variant value is empty `getValueFromParent` is called. But when the omit mandatory check is set to true, saving the variant results in the following error:

`Too few arguments to function Pimcore\Model\DataObject\AbstractObject::get()`

In `getValueFromParent` the class property `$fieldname` is empty. So there is no valid 'containerGetter'.

```php
    public function getValueFromParent($key)
    {
        $object = $this->getObject();
        if ($object) {
            $parent = DataObject\Service::hasInheritableParentObject($object);

            if (!empty($parent)) {
                $containerGetter = 'get' . ucfirst($this->fieldname);
                $brickGetter = 'get' . ucfirst($this->getType());
                $getter = 'get' . ucfirst($key);

                if ($parent->$containerGetter()->$brickGetter()) {
                    return $parent->$containerGetter()->$brickGetter()->$getter();
                }
            }
        }

        throw new InheritanceParentNotFoundException('No parent object available to get a value from');
    }
```

### Steps to reproduce
1. Create object and set inheritance check, add object brick field
2. Create object brick
3. Create the 'main' data object through the Pimcore admin
4. Use the PHP API to create a new variant for the object you created in step 3 (make sure to setOmitMandatoryCheck to true)

### Code example
```php
        $parent = Product::getById(15);

        $new = new Product();
        $new->setKey('test_' . time());
        $new->setParent($parent);
        $new->setType('variant');
        $brick = new AttributesBrick($new);
        $new->getAttributes()->setAttributesBrick($brick);
        $brick->setName('test');

        $new->setOmitMandatoryCheck(true);
        $new->save();
```

**Comment by weisswurstkanone** [comment]
Could you add a unit test please if it is not too much effort?

**Comment by dvesh3** [comment]
resolved by #6672
